sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1813,6 +1813,7 @@ static void _tx_done(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt,
     else {
         /* remove previously held packet */
         gnrc_pktbuf_release(pkt);
+        return;
     }
 #endif /* IS_USED(MODULE_GNRC_NETIF_PKTQ) */
 
```

Comment:
Maybe, just to prevent stuff like this in the future, also add an early/explicit return, in the `if (gnrc_netif_netdev_new_api(netif)) {` block?